### ISSUE TITLE
iotile-gateway broken on py3.5

### ISSUE THREAD

**Issue by mattrunchey** [iotile/coretools#697]
Tests seemed to be hanging. On further investigation, culprit seemed to be in the base64 loader:
```
>>> import pytest
import base64
import json
from iotile.core.hw.reports import BroadcastReport
from iotilegateway import IOTileGateway
from iotile.core.hw import HardwareManager

>>> import base64
>>> import json
>>> from iotile.core.hw.reports import BroadcastReport
>>> from iotilegateway import IOTileGateway
>>> from iotile.core.hw import HardwareManager
>>> 
>>> realtime_config = {
...     "iotile_id": 10,
...     "streams":
...     {
...         "0x5001": [0.001, 100],
...         "0x100a": [0.001, 200]
...     },
... 
...     "broadcast":
...     {
...         "0x1001": [0.020, 100],
...         "0x2001": [0.020, 200]
...     }
... }
>>> 
>>> enc = base64.b64encode(json.dumps(realtime_config).encode('utf-8')).decode('utf-8')
>>> 
>>> config = {
...     'agents': [
...         {
...             "name": "websockets",
...             "args":
...             {
...                 "port": "unused"
...             }
...         }
...     ],
... 
...     'adapters': [
...         {
...             "name": "virtual",
...             "port": "realtime_test@#" + enc + ";simple"
...         }
...     ]
... }
>>>     
KeyboardInterrupt
>>> gateway = IOTileGateway(config)
>>> gateway.start()
>>> Could not load device adapter virtual, quitting
Traceback (most recent call last):
  File "/Users/mrunchey/gitrepos/coretools/iotilegateway/iotilegateway/gateway.py", line 123, in run
    adapter = adapter_class(port_string)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/virtualadapter.py", line 137, in __init__
    loaded_dev = self._load_device(name, config)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/virtualadapter.py", line 185, in _load_device
    config_dict = json.loads(config_str)
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

Will fix this in the py3 migration (wrapping the json.loads input with a utf-8 encoding fixes this).

Interestingly, this only fails on py3.5 -- 3.6 and 3.7 get past this.

**Comment by DimaWittmann** [comment]
Just saw this issue, 
It could be also helpful to add additional assert to the test so next time it won't hang but just fail
`
diff --git a/iotilegateway/test/test_gateway.py b/iotilegateway/test/test_gateway.py
index f74635b..abcdd68 100644
--- a/iotilegateway/test/test_gateway.py
+++ b/iotilegateway/test/test_gateway.py
@@ -51,7 +51,7 @@ def running_gateway():
     gateway = IOTileGateway(config)
 
     gateway.start()
-    gateway.loaded.wait(2.0)
+    assert gateway.loaded.wait(2.0)
 
     hw = HardwareManager(port="ws:127.0.0.1:%d/iotile/v1" % gateway.agents[0].port)
`

**Comment by mattrunchey** [comment]
It's not a bad idea, but `iotile-gateway` is going to be changing significantly soon and I'll need to revamp the tests then (moving from threading to coroutines, so the `wait` function won't be around). 

For the time being, this issue is resolved in #692 .

**Comment by mattrunchey** [comment]
Fixed by #692
